packages/grpc-js/src/server-call.ts

```diff
@@ -428,10 +432,20 @@ export class Http2ServerCallStream<
           ' stream closed with rstCode ' +
           this.stream.rstCode
       );
-      this.cancelled = true;
-      this.emit('cancelled', 'cancelled');
-      this.emit('streamEnd', false);
-      this.sendStatus({code: Status.CANCELLED, details: 'Cancelled by client', metadata: new Metadata()});
+
+      if (!this.statusSent) {
+        this.cancelled = true;
+        this.emit('cancelled', 'cancelled');
+        this.emit('streamEnd', false);
+        this.sendStatus({
+          code: Status.CANCELLED,
+          details: 'Cancelled by client',
+          metadata: null,
+        });
+      }
+
+      // to compensate for a fact that cancelled is not always called
+      this.emit('close');
```

Comment:
I don't understand this change. I am not aware of any code that is listening for this `close` event. In addition, in what situation would an application care about this event if it has already sent a status.

Reply:
<img width="746" alt="Screen Shot 2022-10-19 at 15 21 00" src="https://user-images.githubusercontent.com/1713617/196815638-7fb2832e-5dff-48e5-8680-8aa60d6153f4.png">

<img width="741" alt="Screen Shot 2022-10-19 at 15 21 28" src="https://user-images.githubusercontent.com/1713617/196815688-5ba73c70-eadb-4bf2-b673-7b6cdee96d71.png">

Reply:
Reverted this and if needed we can do a follow up on this - so it's just a separate unrelated issue

Reply:
If you're handling a stream of messages from the remote end, you should be listening for the `end` event to know when you have received all messages. And conversely, if you are sending a stream of messages, you should call the `end` event to indicate that you have finished writing. You shouldn't generally need to listen for the `cancelled` event unless you are doing a long asynchronous operation. I don't know why you would need to delete the `call` at all.